### PR TITLE
Hierarchy refactoring of docs

### PR DESCRIPTION
**Pull request by sarahmonster**
This is an attempt to make the styleguide a bit more usable & readable. It may or may not look super different.

Screenshot 2019-07-05 at 23 59 58

I'd kinda love it if we could put an intro paragraph under the title, _then_ the "props" drop-down, but that may be harder. The props table itself still needs some work, but it would help to get your @tofumatt's opinion from a developer perspective about how we could make it more readable. (I have some thoughts, but I'd love to know how it's working for you since I expect you might  look at this more than me!)

A big improvement I think we could make here is making the sidebar navigation more self-contained—so maybe a slide-in (and out) element that would default to open on large screens, and closed for smaller ones. Something along these lines: https://tympanus.net/Blueprints/SlidePushMenus/  I've also been trying to add selected styling, but I've been struggling thanks to #106.

We may also want to figure out a way of implenting sticky Tabs (available as of #108!) to make navigating between the major sections easier, but I suspect that's going to be a bit tricky.

Probably all of the above I can file as separate issues, and we can go ahead and merge this now.

### CODE REVIEW

File: styleguide/components/SectionHeading/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import theme from 'styleguide/theme';
+
+function SectionHeadingRenderer({ children, toolbar, id, href, depth, deprecated }) {
+	// const headingLevel = Math.min(6, depth);
+	// Overwrite RSG's classification of headings, since we're only using top-level headings here!
+	const headingLevel = 1;
+
+	return (
+			<Heading level={headingLevel} id={id}>
```
Comment (tofumatt) [inline styleguide/components/SectionHeading/index.js]: Geeez what is happening in here with your indentation? 😨

Comment (sarahmonster) [reply styleguide/components/SectionHeading/index.js]: Probably it's copied from the RSG files directly on Github?

INSTALL A LINTER PLS.

Comment (sarahmonster) [reply styleguide/components/SectionHeading/index.js]: (Tell me if that's fixed it now? I can't easily tell!)

**Comment by tofumatt** [comment]
That component looks better but the rest are off. Let's just ignore that and we'll start linting that folder in the future. 🚢

**Pull request merged**